docs/releasing/README.md

```diff
@@ -0,0 +1,182 @@
+                "AVALANCHEGO_VERSION": "v1.13.0"
+            }
+        },
+    }
+    ```
+
+1. Add an entry in the object in [compatibility.json](../../compatibility.json), adding the target release `$VERSION` as key and the AvalancheGo RPC chain VM protocol version as value, to the `"rpcChainVMProtocolVersion"` JSON object. For example, we would add:
+
+    ```json
+    "v0.4.0": 39,
+    ```
+
+    💁 If you are unsure about the RPC chain VM protocol version, set the version to `0`, for exampple `"v0.4.0": 0`, and then run:
+
+    ```bash
+    go test -run ^TestCompatibility$ github.com/ava-labs/precompile-evm/plugin
+    ```
+
+    This will fail with an error similar to:
+
+    ```text
+    compatibility.json has precompile-evm version v0.4.0 stated as compatible with RPC chain VM protocol version 0 but AvalancheGo protocol version is 39
+    ```
+
+    This message can help you figure out what the correct RPC chain VM protocol version (here `39`) has to be in compatibility.json for your current release.
+1. Specify the AvalancheGo compatibility in the [README.md relevant section](../../README.md#avalanchego-compatibility). For example we would add:
+
+    ```text
+    ...
+    [v0.4.0] AvalancheGo@v1.12.2/1.13.0-fuji/1.13.0 (Protocol Version: 39)
+    ```
+
+1. Commit your changes and push the branch
+
+    ```bash
+    git add .
+    git commit -S -m "chore: release $VERSION"
+    git push -u origin "$GITHUB_USER/releases/$VERSION"
+    ```
+
+1. Create a pull request (PR) from your branch targeting main, for example using [`gh`](https://cli.github.com/):
+
+    ```bash
+    gh pr create --repo github.com/ava-labs/precompile-evm --base main --title "chore: release $VERSION"
+    ```
+
+1. Wait for the PR checks to pass
+1. Update the `hello-world-example` branch to be rebased on your release branch:
+
+    ```bash
+    git fetch origin hello-world-example:hello-world-example
+    git checkout hello-world-example
+    git rebase "$GITHUB_USER/releases/$VERSION"
+    # Fix eventual conflicts
+    git push --force hello-world-example
+    ```
+
+1. [Wait for the checks](https://github.com/ava-labs/precompile-evm/pull/12/checks) of the `hello-world-example` branch [PR](https://github.com/ava-labs/precompile-evm/pull/12) to pass. **Never merge this PR**. You can also use `gh` with:
+
+    ```bash
+    gh pr checks 12 --watch
+    ```
+
+1. Squash and merge your release branch into `main`, for example:
+
+    ```bash
+    gh pr merge "$GITHUB_USER/releases/$VERSION" --squash --delete-branch --subject "chore: release $VERSION" --body "\n- Bump subnet-evm from v0.7.3 to v0.7.4\n- Update AvalancheGo from v1.12.3 to v1.13.0"
+    ```
+
+1. Create a release branch from the `main` branch, this time without your username prefix:
+
+    ```bash
+    git checkout -b "releases/$VERSION"
+    ```
+
+    This is to avoid creating a release targeting the `main` branch, which may contain new commits merged whilst the release is being created.
```

Comment:
For precompile-evm in particular, we could instead create and push a tag, and then create the release from it.

However, for subnet-evm (and others), since goreleaser is triggered on a tag push, this would fail since it would try to find the Github release which may not be created in time after the tag gets pushed. To keep a bit of a similar release process across repositories, we can either:

- keep it with these branches `releases/$VERSION` to target the release to
- change the goreleaser job to trigger on **release creation** and not on **tag creation** - I think that one would be interesting, thoughts on this?

Regarding the new "recent commit" tab you can pick from, I would argue it's rather error prone to pick the right commit.

Reply:
I agree it's annoying that goreleaser can do releases it's own. There is a release trigger in actions. If we can configure it to trigger in case we publish a release or prerelease (but not on draft save/create) then it would work.

Reply:
Changed to create a tag from main and then create a release using that tag.
In this repo there is no goreleaser, but I'll change it to trigger on releases only on subnet-evm and coreth